sdk/communication/communication-sms/test/smsClient.mocked.spec.ts

```diff
@@ -45,73 +77,153 @@ describe("[mocked] SmsClient", async () => {
       to: ["+18768985487"],
       message: "message"
     };
-    const { _response } = await smsClient.send(sendRequest);
 
+    const _responses = await smsClient.send(sendRequest);
+    const _response = _responses[0];
     sinon.assert.calledOnce(spy);
-    assert.equal(_response.status, 200);
-    assert.equal(_response.parsedBody.messageId, "id");
-
+    assert.equal(_response.httpStatusCode, 202);
+    assert.equal(_response.messageId, "id");
     const request = spy.getCall(0).args[0];
-
     assert.equal(request.url, `${baseUri}/sms?api-version=${API_VERSION}`);
     assert.equal(request.method, "POST");
-    assert.deepEqual(JSON.parse(request.body), { ...sendRequest, sendSmsOptions: {} });
+
+    const expectedRequestBody = {
+      from: sendRequest.from,
+      smsRecipients: [
+        {
+          to: "+18768985487",
+          repeatabilityFirstSent: new Date().toUTCString(),
+          repeatabilityRequestId: mockedGuid
+        }
+      ],
+      message: sendRequest.message,
+      smsSendOptions: {
+        enableDeliveryReport: false
+      }
+    };
+
+    assert.deepEqual(JSON.parse(request.body), expectedRequestBody);
   });
 
   it("sets correct headers", async () => {
     smsClient = new SmsClient(baseUri, new AzureKeyCredential("banana"), {
       httpClient: mockHttpClient
     });
     const spy = sinon.spy(mockHttpClient, "sendRequest");
-    const sendRequest: SendRequest = {
+    const sendRequest: SmsSendRequest = {
       from: "+18768984505651",
       to: ["+18768985487"],
       message: "message"
     };
-    const { _response } = await smsClient.send(sendRequest);
-
+    const _responses = await smsClient.send(sendRequest);
+    const _response = _responses[0];
     sinon.assert.calledOnce(spy);
-    assert.equal(_response.status, 200);
-
+    assert.equal(_response.httpStatusCode, 202);
     const request = spy.getCall(0).args[0];
-
     if (isNode) {
       assert.equal(request.headers.get("host"), "contoso.api.fake:443");
     }
-
     assert.typeOf(request.headers.get(dateHeader), "string");
     assert.isDefined(request.headers.get("authorization"));
     assert.match(
       request.headers.get("authorization") as string,
       /HMAC-SHA256 SignedHeaders=.+&Signature=.+/
     );
   });
+
+  it("retries with same repeatability id when service is unreachable", async () => {
+    const smsClient = new SmsClient(baseUri, new AzureKeyCredential("banana"), {
+      httpClient: mockFailingHttpClient
+    });
+    const spy = sinon.spy(mockFailingHttpClient, "sendRequest");
+    const sendRequest: SmsSendRequest = {
+      from: "+18768984505651",
+      to: ["+18768985487"],
+      message: "message"
+    };
+
+    const clock = sinon.useFakeTimers();
+    const expectedRequestBody = {
+      from: sendRequest.from,
+      smsRecipients: [
+        {
+          to: "+18768985487",
+          repeatabilityFirstSent: new Date().toUTCString(),
+          repeatabilityRequestId: mockedGuid
```

Comment:
Well you're not really testing that we use the same GUID in retries because you mocked the guid generator to always return the same GUID.

To really test it, you would need to mock so that it returns GUID1, GUID2, ...

But it's probably easier to have a spy on UUID and assert that it's only getting called once.

Reply:
there's also some test reliability issues with the fake timer and the retry. I'm thinking of removing this test altogether. Instead going to have a different test to test the guids.

Reply:
removed for now, a version will exist in a new PR